examples/callbacks.jl

```diff
@@ -28,12 +28,14 @@ function example_lazy_constraint()
         lazy_called = true
         x_val = callback_value(cb_data, x)
         y_val = callback_value(cb_data, y)
-        if y_val - x_val > 1 + 1e-6
-            con = @build_constraint(y - x <= 1)
-            MOI.submit(model, MOI.LazyConstraint(cb_data), con)
-        elseif y_val + x_val > 3 + 1e-6
-            con = @build_constraint(y - x <= 1)
-            MOI.submit(model, MOI.LazyConstraint(cb_data), con)
+        if (isinteger.(x_val) == true) && (isinteger.(y_val) == true) # Add the cut only on integral solutions
```

Comment:
```Julia
if isinteger(x_val) && isinteger(y_val)
```

Reply:
Using `isinteger` isn't correct. If you don't use the same integrality tolerance as the solver, then you might fail to add cuts on a solution that the solver thinks is integral. Maybe we need to provide better support for adding a lazy constraint callback that is called only at integer solutions. We previously had the `fractional` keyword argument: 
https://github.com/JuliaOpt/JuMP.jl/blob/a9de4d0de1b34b85d35c3624bea236afec7e5bec/src/callbacks.jl#L32

Reply:
How about a `callback_status(cb_data)` which returns an enum `CBNODE_FRACTIONAL`, `CBNODE_INTEGER`, or `CBNODE_UNKNOWN`.

Reply:
I haven't thought through all the implications, but that could work also.

Reply:
+1 for something like `MOI.CallbackNodeStatus`

Reply:
I agree that `isinteger` was just a hack for that simple example which was there. I actually wrote my own function to handle the integrality tolerance for my problem. It would be ideal to invoke lazy callback at only integral solutions, which was also my original concern.

Reply:
I think callback function form `function callback_function(cb_data, cb_where)` can be a proper choice, `cb_where` can be used to control where we can add constraints.

Reply:
@YP-Ye, currently `cb_where` is for solver-dependent callbacks. We're looking at introducing a way for solver-independent callbacks (which only take `(cb_data)`) to identify where they are being called from.